Puc/v4p4/UpdateChecker.php

```diff
@@ -309,6 +311,48 @@ protected function filterUpdateResult($update, $httpResult = null) {
 			return $update;
 		}
 
+		/**
+		 * The "Tested up to" field in the plugin metadata is supposed to be in the form of "major.minor",
+		 * while WordPress core's list_plugin_updates() expects the $update->tested field to be an exact
+		 * version, e.g. "major.minor.patch", to say it's compatible. In other case it shows
+		 * "Compatibility: Unknown".
+		 * The function mimics how wordpress.org API crafts the "tested" field out of "Tested up to".
+		 *
+		 * @param Puc_v4p4_Update|null $update
+		 */
+		private function fixSupportedWordpressVersion(Puc_v4p4_Update $update = null) {
```

Comment:
Please change this to `protected` so that people have the option to disable it or to substitute their own algorithm.